### PR TITLE
Add data checker workflow and some data checkers

### PR DESCRIPTION
**Pull request by orowith2os**
Supersedes https://github.com/flathub/shared-modules/pull/237 and https://github.com/flathub/shared-modules/pull/238

Allows closing https://github.com/flathub/shared-modules/issues/92

Would like a review from (2-3) others before merging.

### CODE REVIEW

File: update-checker.txt
Comment (barthalion) [inline update-checker.txt:1]: There's probably no need to make this opt-in. Just grep for x-data-checker across the repo to get the list of modules that "signed up".

**Review by wjt — COMMENTED** [review]
Very nice! I Agree With Bart.

File: .github/workflows/update.yaml
Comment (ranisalt) [inline .github/workflows/update.yaml:4]:
```suggestion
    - cron: "0 * * * *" # run every hour
```

To keep the formatting consistent

File: update-checker.sh
Comment (orowith2os) [reply update-checker.sh:28]: The pwd? It should be, from what I've seen with other workflows and whatnot. 

As Bart mentioned, I should probably just grep for the data checker instead, so I can get the *real* paths and not trust this file.

**Comment by orowith2os** [comment]
Oh dear, the commit history is a mess. I'll make sure to squash these when I get some approvals.

**Comment by orowith2os** [comment]
I've tested with gzdoom and the SDL2 manifests, all appears to be working well.

**Comment by ranisalt** [comment]
> Oh dear, the commit history is a mess. I'll make sure to squash these when I get some approvals.

Don't worry, it can be done by the merger. Great job!

**Comment by orowith2os** [comment]
@barthalion @hfiguiere @wjt lil reminder, would you mind giving me your thoughts on this? Am I good to merge?

File: README.md
Comment (barthalion) [inline README.md:50]: This is out of date now, isn't it?

File: .github/workflows/update.yaml
Comment (wjt) [inline .github/workflows/update.yaml:32]: I think this argument is now unneeded?

File: .github/workflows/update.yaml
Comment (wjt) [inline .github/workflows/update.yaml:31]: Does this actually work? Do you not need:

```suggestion
          entrypoint: ./update-checker.sh
```

(I have no idea either way.)

**Review by wjt — COMMENTED** [review]
Seems OK, except for a nitpick similar to Bart's

File: update-checker.sh
Comment (wjt) [inline update-checker.sh:26]:
```
In update-checker.sh line 26:
  echo "Running data checker on: $line"
                                 ^---^ SC2154: line is referenced but not assigned.
```

File: update-checker.sh
Comment (wjt) [inline update-checker.sh:31]:
```
In update-checker.sh line 31:
   flatpak run --filesystem="$(pwd)" org.flathub.flatpak-external-data-checker ${args[@]} $path
                                                                               ^--------^ SC2068: Double quote array expansions to avoid re-splitting elements.
                                                                                          ^---^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
   flatpak run --filesystem="$(pwd)" org.flathub.flatpak-external-data-checker ${args[@]} "$path"
```

Comment (wjt) [reply update-checker.sh:31]: I think what you want is:

```suggestion
   flatpak run --filesystem="$(pwd)" org.flathub.flatpak-external-data-checker "${args[@]}" "$path"
```

File: update-checker.sh
Comment (wjt) [inline update-checker.sh:33]:
```
In update-checker.sh line 33:
   /app/flatpak-external-data-checker ${args[@]} $path
                                      ^--------^ SC2068: Double quote array expansions to avoid re-splitting elements.
                                                 ^---^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
   /app/flatpak-external-data-checker ${args[@]} "$path"
```

```suggestion
   /app/flatpak-external-data-checker "${args[@]}" "$path"
```

File: update-checker.sh
Comment (wjt) [reply update-checker.sh:26]:
```suggestion
  echo "Running data checker on: $path"
```

**Review by wjt — CHANGES_REQUESTED** [review]
I ran Shellcheck over the script.

For more information:
  https://www.shellcheck.net/wiki/SC2068 -- Double quote array expansions to ...
  https://www.shellcheck.net/wiki/SC2154 -- line is referenced but not assign...
  https://www.shellcheck.net/wiki/SC2086 -- Double quote to prevent globbing ...

File: .github/workflows/update.yaml
Comment (orowith2os) [reply .github/workflows/update.yaml:31]: I think what it already has is correct, going off of the GitHub docs?

**Comment by orowith2os** [comment]
Shellcheck says it's okay, I've updated the README, and the workflow appears to be correct, going off of the [GitHub docs](https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#runsentrypoint), I think?

File: .github/workflows/update.yaml
Comment (wjt) [reply .github/workflows/update.yaml:31]: OK!

**Review by wjt — APPROVED** [review]
I don't have commit access, but this LGTM

**Comment by orowith2os** [comment]
I think this is going to need more thought and work, and probably some porting to Python - I want to account for cases where submodules rely on generated dependencies too, and it's also possible that the layout of some submodules isn't what this checker expects. 

This could also do with some work in the main external-data-checker too.

**Comment by bbhtt** [comment]
I would merge this as is now. I don't mind having it in shell and future improvements or python ports can be done later.

Two things - 

1.  Commits should be squashed
2. The workflow run should be weekly, x-checker doesn't have super active maintenance and I don't think piling PRs is a good idea.

**Comment by bbhtt** [comment]
Handled both, let's merge this.

**Comment by bbhtt** [comment]
Seems to be working after a minor fix: https://github.com/flathub/shared-modules/commit/8d84ae268a653163f689e41d9ab441a671754510

https://github.com/flathub/shared-modules/actions/runs/11093647389

These were the first two PRs:

https://github.com/flathub/shared-modules/pull/321

https://github.com/flathub/shared-modules/pull/320

**Pull request merged**
